### DISCUSSION TITLE
Reseting the envelope controls

### DISCUSSION THREAD

**Discussion by nidhiwebsolutions** [katspaugh/wavesurfer.js#2962] [Q&A] [answered]
I want to reset the fade-in fade-out controls to initial stage, below code only changes start and end time

```
activeWsEnvelope.setStartTime(0, true);
activeWsEnvelope.setEndTime(30, true);
```
If I do this

```
activeWsEnvelope.destroy();
activeWsEnvelope = trimws.registerPlugin(WaveSurfer.Envelope.create(envolpeOptions()));
```
it gives following error in chrome, but works in firefox, I am using latest version

> Uncaught DOMException: Failed to execute 'createMediaElementSource' on 'AudioContext': HTMLMediaElement already connected previously to a different MediaElementSourceNode

**Comment by katspaugh** [comment, marked as answer]
Please feel free to add `setFadeInEnd` and `setFadeOutStart` and propose a PR.

**Reply by nidhiwebsolutions** [reply]
Can there be a new parameter in these functions

```
setStartTime(time, moveFadeInEnd?): void
setEndTime(time, moveFadeOutStart?): void
```

**Reply by katspaugh** [reply]
There's a param, but it moves those fade points to the same distance as the start/end points are moved, so it's not what you want, I assume.
